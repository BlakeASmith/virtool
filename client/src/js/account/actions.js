/**
 *
 *
 * @copyright 2017 Government of Canada
 * @license MIT
 * @author igboyes
 *
 */
import {
    WS_UPDATE_ACCOUNT,
    GET_ACCOUNT,
    UPDATE_ACCOUNT_SETTINGS,
    CREATE_API_KEY,
    UPDATE_API_KEY,
    REMOVE_API_KEY,
    LOGOUT
} from "../actionTypes";


export function wsUpdateAccount (data) {
    return {
        type: WS_UPDATE_ACCOUNT,
        data
    };
}

export function getAccount () {
    return {
        type: GET_ACCOUNT.REQUESTED
    }
}

export function updateAccountSettings (update) {
    return {
        type: UPDATE_ACCOUNT_SETTINGS.REQUESTED,
        update
    };
}

export function createAPIKey (name, permissions, callback) {
    return {
        type: CREATE_API_KEY.REQUESTED,
        name,
        permissions,
        callback
    };
}

export function updateAPIKey (keyId, permissions) {
    return {
        type: UPDATE_API_KEY.REQUESTED,
        keyId,
        permissions
    };
}

export function removeAPIKey (keyId) {
    return {
        type: REMOVE_API_KEY.REQUESTED,
        keyId
    };
}

export function logout () {
    return {
        type: LOGOUT.REQUESTED
    }
}
