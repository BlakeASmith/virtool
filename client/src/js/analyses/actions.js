import {
    WS_INSERT_ANALYSIS,
    WS_UPDATE_ANALYSIS,
    WS_REMOVE_ANALYSIS,
    FIND_ANALYSES,
    GET_ANALYSIS,
    ANALYZE,
    BLAST_NUVS,
    REMOVE_ANALYSIS,
    CLEAR_ANALYSIS,
    COLLAPSE_ANALYSIS,
    SET_PATHOSCOPE_FILTER,
    TOGGLE_SORT_PATHOSCOPE_DESCENDING,
    TOGGLE_SHOW_PATHOSCOPE_READS,
    SET_ACTIVE_HIT_ID,
    TOGGLE_FILTER_ORFS,
    TOGGLE_FILTER_SEQUENCES,
    SET_SEARCH_IDS,
    SET_ANALYSIS_SORT_KEY,
    TOGGLE_RESULT_EXPANDED,
    CLEAR_ANALYSES
} from "../app/actionTypes";
import { simpleActionCreator } from "../utils/utils";

export const wsInsertAnalysis = data => ({
    type: WS_INSERT_ANALYSIS,
    data
});

/**
 * Returns an action that should be dispatched when a analysis document is updated via websocket.
 *
 * @func
 * @param data {object} update data passed in the websocket message
 * @returns {object}
 */
export const wsUpdateAnalysis = data => ({
    type: WS_UPDATE_ANALYSIS,
    data
});

/**
 * Returns an action that should be dispatched when a analysis document is removed via websocket.
 *
 * @func
 * @param data {string} the id for the specific analysis
 * @returns {object}
 */
export const wsRemoveAnalysis = data => ({
    type: WS_REMOVE_ANALYSIS,
    data
});

export const collapseAnalysis = simpleActionCreator(COLLAPSE_ANALYSIS);

export const setActiveHitId = id => ({
    type: SET_ACTIVE_HIT_ID,
    id
});

export const setSearchIds = ids => ({
    type: SET_SEARCH_IDS,
    ids
});

export const setAnalysisSortKey = sortKey => ({
    type: SET_ANALYSIS_SORT_KEY,
    sortKey
});

export const setPathoscopeFilter = key => ({
    type: SET_PATHOSCOPE_FILTER,
    key
});

export const toggleFilterORFs = simpleActionCreator(TOGGLE_FILTER_ORFS);

export const toggleFilterSequences = simpleActionCreator(TOGGLE_FILTER_SEQUENCES);

export const togglePathoscopeSortDescending = simpleActionCreator(TOGGLE_SORT_PATHOSCOPE_DESCENDING);

export const toggleResultExpanded = id => ({
    type: TOGGLE_RESULT_EXPANDED,
    id
});

export const toggleShowPathoscopeReads = simpleActionCreator(TOGGLE_SHOW_PATHOSCOPE_READS);

export const findAnalyses = (sampleId, term, page) => ({
    type: FIND_ANALYSES.REQUESTED,
    sampleId,
    term,
    page
});

/**
 * Returns action that can trigger an API call for retrieving a specific analysis.
 *
 * @func
 * @param analysisId {string} unique analysis id
 * @returns {object}
 */
export const getAnalysis = analysisId => ({
    type: GET_ANALYSIS.REQUESTED,
    analysisId
});

export const clearAnalyses = simpleActionCreator(CLEAR_ANALYSES);
export const clearAnalysis = simpleActionCreator(CLEAR_ANALYSIS);

/**
 * Returns action that can trigger an API call for sample analysis.
 *
 * @func
 * @param sampleId {string} unique sample id
 * @param refId {string} unique id for a reference
 * @param algorithm {string} algorithm type
 * @param userId {string} the id of the requesting user
 * @returns {object}
 */
export const analyze = (sampleId, refId, algorithm, subtractionId, userId) => ({
    type: ANALYZE.REQUESTED,
    algorithm,
    refId,
    sampleId,
    subtractionId,
    userId
});

/**
 * Returns action that can trigger an API call for BLASTing NuV analysis contigs.
 *
 * @func
 * @param analysisId {string} unique analysis id
 * @param sequenceIndex {number} index of the sequence
 * @returns {object}
 */
export const blastNuvs = (analysisId, sequenceIndex) => ({
    type: BLAST_NUVS.REQUESTED,
    analysisId,
    sequenceIndex
});

/**
 * Returns action that can trigger an API call for removing an analysis.
 *
 * @func
 * @param analysisId {string} unique analysis id
 * @returns {object}
 */
export const removeAnalysis = analysisId => ({
    type: REMOVE_ANALYSIS.REQUESTED,
    analysisId
});
