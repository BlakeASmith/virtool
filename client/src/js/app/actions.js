import { PUSH_STATE } from "./actionTypes";

export const pushState = state => ({
    type: PUSH_STATE,
    state
});
