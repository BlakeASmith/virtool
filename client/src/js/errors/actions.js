/**
 * Redux action creator for clearing errors.
 *
 * @module error/actions
 */
import { CLEAR_ERROR } from "../actionTypes";

/**
 * Clears specific temprorarily stored error.
 *
 * @func
 * @returns {object}
 */
export const clearError = (error) => ({
    type: CLEAR_ERROR,
    error
});
