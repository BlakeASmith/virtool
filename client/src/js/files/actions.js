/**
 * Redux action creators for use with file management.
 *
 * @module files/actions
 */

import { simpleActionCreator } from "../utils";
import {
    WS_UPDATE_FILE,
    WS_REMOVE_FILE,
    FIND_FILES,
    REMOVE_FILE,
    UPLOAD,
    UPLOAD_PROGRESS,
    HIDE_UPLOAD_OVERLAY
} from "../actionTypes";

/**
 * Returns an action that should be dispatched when a file document is updated via websocket.
 *
 * @func
 * @param data {object} the data passed in the websocket message
 * @returns {object}
 */
export const wsUpdateFile = (data) => ({
    type: WS_UPDATE_FILE,
    data
});

/**
 * Returns an action that should be dispatched when a file document is removed via websocket.
 *
 * @func
 * @param data {object} the data passed in the websocket message
 * @returns {object}
 */
export const wsRemoveFile = (data) => ({
    type: WS_REMOVE_FILE,
    data
});

/**
 * Returns an action that can trigger an API call that finds file documents.
 *
 * @func
 * @param fileType {string} a file type to filter the returned documents by
 * @param page {number} which page of results to return
 * @returns {object}
 */
export const findFiles = (fileType, page) => ({
    type: FIND_FILES.REQUESTED,
    fileType,
    page
});

/**
 * Returns an action that can trigger the upload of a file to the server.
 *
 * @param localId
 * @param file
 * @param fileType
 * @param onProgress
 * @returns {{type: string, localId: *, file: *, fileType: *, onProgress: *}}
 */
export const upload = (localId, file, fileType, onProgress) => ({
    type: UPLOAD.REQUESTED,
    localId,
    file,
    fileType,
    onProgress
});

export const removeFile = (fileId) => ({
    type: REMOVE_FILE.REQUESTED,
    fileId
});

export const uploadProgress = (localId, progress) => ({
    type: UPLOAD_PROGRESS,
    localId,
    progress
});

export const hideUploadOverlay = simpleActionCreator(HIDE_UPLOAD_OVERLAY);
