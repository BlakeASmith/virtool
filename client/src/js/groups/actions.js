import { simpleActionCreator } from "../utils";
import { LIST_GROUPS, CREATE_GROUP, SET_GROUP_PERMISSION, REMOVE_GROUP } from "../actionTypes";

export const listGroups = simpleActionCreator(LIST_GROUPS.REQUESTED);

export const createGroup = (groupId) => ({
    type: CREATE_GROUP.REQUESTED,
    groupId
});

export const setGroupPermission = (groupId, permission, value) => ({
    type: SET_GROUP_PERMISSION.REQUESTED,
    groupId,
    permission,
    value
});

export const removeGroup = (groupId) => ({
    type: REMOVE_GROUP.REQUESTED,
    groupId
});
