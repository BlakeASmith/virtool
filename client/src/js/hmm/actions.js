/**
 *
 *
 * @copyright 2017 Government of Canada
 * @license MIT
 * @author igboyes
 *
 */
import { FIND_HMMS, GET_HMM, INSTALL_HMMS } from "../actionTypes";

export const findHMMs = (term, page) => {
    return {
        type: FIND_HMMS.REQUESTED,
        term,
        page
    };
};

export const getHmm = (hmmId) => {
    return {
        type: GET_HMM.REQUESTED,
        hmmId: hmmId
    };
};

export const installHMMs = () => {
    return {
        type: INSTALL_HMMS.REQUESTED
    }
};
