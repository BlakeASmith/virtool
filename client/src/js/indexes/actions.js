import { simpleActionCreator } from "../utils";
import {
    FIND_INDEXES,
    GET_INDEX,
    GET_UNBUILT,
    CREATE_INDEX,
    GET_INDEX_HISTORY,
    SHOW_REBUILD,
    HIDE_REBUILD
} from "../actionTypes";

export const findIndexes = simpleActionCreator(FIND_INDEXES.REQUESTED);

export const getIndex = (indexVersion) => ({
    type: GET_INDEX.REQUESTED,
    indexVersion
});

export const getUnbuilt = simpleActionCreator(GET_UNBUILT.REQUESTED);

export const createIndex = simpleActionCreator(CREATE_INDEX.REQUESTED);

export const getIndexHistory = (indexVersion) => ({
    type: GET_INDEX_HISTORY.REQUESTED,
    indexVersion
});

export const showRebuild = () => simpleActionCreator(SHOW_REBUILD);

export const hideRebuild = simpleActionCreator(HIDE_REBUILD);
