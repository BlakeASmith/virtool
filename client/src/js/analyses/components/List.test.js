import AnalysesListContainer, { AnalysesToolbar } from "./List";
import * as actions from "../actions";
import * as indexesActions from "../../indexes/actions";
import * as hmmActions from "../../hmm/actions";

describe("<AnalysesList />", () => {
    const initialState = {
        account: {
            administrator: false,
            id: "test-user",
            groups: []
        },
        analyses: {
            sampleId: "test-sample",
            documents: [],
            filter: "search",
            readyIndexes: []
        },
        samples: {
            detail: { all_write: false, group_write: false, group: "" }
        },
        hmms: { status: { installed: false } }
    };
    const store = mockStore(initialState);
    let props;
    let wrapper;

    it("renders correctly", () => {
        wrapper = shallow(<AnalysesListContainer store={store} />).dive();
        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({
            analyses: [{ timestamp: "2018-01-01T00:00:00.000000Z" }],
            hmms: { status: { installed: true } }
        });
        expect(wrapper).toMatchSnapshot();
    });

    it("renders <LoadingPlaceholder /> when analyses, hmms, or indexeses data are null", () => {
        wrapper = shallow(<AnalysesListContainer store={store} />).dive();
        wrapper.setProps({ indexes: null });
        expect(wrapper).toMatchSnapshot();
    });

    it("Clicking on create button or closing modal changes modal's show state", () => {
        wrapper = shallow(<AnalysesListContainer store={store} />).dive();
        expect(wrapper.state("show")).toBe(false);

        wrapper.find({ term: "search" }).prop("onClick")();
        expect(wrapper.state("show")).toBe(true);

        wrapper.find({ userId: "test-user" }).prop("onHide")();
        expect(wrapper.state("show")).toBe(false);
    });

    it("renders <AnalysesToolbar /> subcomponent", () => {
        props = {
            term: "test",
            isDisabled: false,
            onClick: jest.fn(),
            onFilter: jest.fn()
        };
        wrapper = shallow(<AnalysesToolbar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    describe("Dispatch Functions", () => {
        let spy;

        it("Entering search input dispatches filterAnalyses() action", () => {
            spy = sinon.spy(actions, "filterAnalyses");
            expect(spy.called).toBe(false);

            wrapper = shallow(<AnalysesListContainer store={store} />).dive();
            wrapper.find({ term: "search" }).prop("onFilter")({ target: { value: "example" } });

            expect(spy.calledWith("test-sample", "example")).toBe(true);

            spy.restore();
        });

        it("Creating a new analysis dispatches analyze() action", () => {
            spy = sinon.spy(actions, "analyze");
            expect(spy.called).toBe(false);

            const sampleId = "test-sample1";
            const references = [{ refId: "test-ref" }];
            const algorithm = "test-algorithm";
            const userId = "test-user";

            wrapper = shallow(<AnalysesListContainer store={store} />).dive();
            wrapper.find({ userId: "test-user" })
                .prop("onSubmit")(sampleId, references, algorithm, userId);

            expect(spy.calledWith(sampleId, references[0].refId, algorithm, userId)).toBe(true);

            spy.restore();
        });

        it("Component mount dispatches listHmms() and listReadyIndexes() actions", () => {
            const spyListIndexes = sinon.spy(indexesActions, "listReadyIndexes");
            const spyListHMMs = sinon.spy(hmmActions, "listHmms");
            expect(spyListIndexes.called).toBe(false);
            expect(spyListHMMs.called).toBe(false);

            wrapper = shallow(<AnalysesListContainer store={store} />).dive();
            expect(spyListIndexes.calledOnce).toBe(true);
            expect(spyListHMMs.calledOnce).toBe(true);

            spyListIndexes.restore();
            spyListHMMs.restore();
        });

    });

});
