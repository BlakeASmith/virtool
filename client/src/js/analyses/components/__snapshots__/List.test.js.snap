// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalysesList /> renders <AnalysesToolbar /> subcomponent 1`] = `
<div
  className="toolbar"
>
  <FormGroup
    bsClass="form-group"
  >
    <InputGroup
      bsClass="input-group"
    >
      <InputGroupAddon
        bsClass="input-group-addon"
      >
        <Icon
          faStyle="fas"
          fixedWidth={false}
          name="search"
          pullRight={false}
        />
      </InputGroupAddon>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        onChange={[MockFunction]}
        placeholder="User or reference"
        type="text"
        value="test"
      />
    </InputGroup>
  </FormGroup>
  <Button
    bsStyle="primary"
    disabled={false}
    icon="plus-square fa-fw"
    onClick={[MockFunction]}
    pullRight={false}
    tip="New Analysis"
    tipPlacement="top"
  />
</div>
`;

exports[`<AnalysesList /> renders <LoadingPlaceholder /> when analyses, hmms, or indexeses data are null 1`] = `
<LoadingPlaceholder
  margin="37px"
/>
`;

exports[`<AnalysesList /> renders correctly 1`] = `
<div>
  <Alert
    bsClass="alert"
    bsStyle="warning"
    closeLabel="Close alert"
  >
    <Flex
      alignContent="stretch"
      alignItems="center"
      direction="row"
      justifyContent="flex-start"
      wrap="nowrap"
    >
      <Icon
        faStyle="fas"
        fixedWidth={false}
        name="info-circle"
        pullRight={false}
      />
      <FlexItem
        alignSelf={null}
        basis="auto"
        grow={0}
        pad={5}
        shrink={1}
      >
        <span>
          The HMM data is not installed. 
        </span>
        <Link
          replace={false}
          to="/hmm"
        >
          Install HMMs
        </Link>
        <span>
           to use in further NuV analyses.
        </span>
      </FlexItem>
    </Flex>
  </Alert>
  <AnalysesToolbar
    isDisabled={true}
    onClick={[Function]}
    onFilter={[Function]}
    term="search"
  />
  <ListGroup
    bsClass="list-group"
  >
    <NoneFound
      noListGroup={true}
      noun="analyses"
    />
  </ListGroup>
  <CreateAnalysis
    hasHmm={false}
    onHide={[Function]}
    onSubmit={[Function]}
    refIndexes={Array []}
    show={false}
    userId="test-user"
  />
</div>
`;

exports[`<AnalysesList /> renders correctly 2`] = `
<div>
  <AnalysesToolbar
    isDisabled={true}
    onClick={[Function]}
    onFilter={[Function]}
    term="search"
  />
  <ListGroup
    bsClass="list-group"
  >
    <Connect(AnalysisItem)
      key="0"
      timestamp="2018-01-01T00:00:00.000000Z"
    />
  </ListGroup>
  <CreateAnalysis
    hasHmm={true}
    onHide={[Function]}
    onSubmit={[Function]}
    refIndexes={Array []}
    show={false}
    userId="test-user"
  />
</div>
`;
